internal/build/lifecycle_execution.go

```diff
@@ -281,7 +281,7 @@ func (l *LifecycleExecution) Run(ctx context.Context, phaseFactoryCreator PhaseF
 		return l.Export(ctx, buildCache, launchCache, kanikoCache, phaseFactory)
 	}
 
-	if l.platformAPI.AtLeast("0.10") && l.hasExtensions() {
+	if l.platformAPI.AtLeast("0.10") && l.hasExtensions() && !l.opts.UseCreatorWithExtensions {
 		return errors.New("builder has an order for extensions which is not supported when using the creator")
```

Comment:
I wonder if we can improve the error message to give some guidance to end-users on what's going on or what they can do to fix it.. maybe a link to some documentation?